### PR TITLE
Manager changes for v7 on-demand store

### PR DESCRIPTION
**Pull request by tmeasday**
Part of https://github.com/storybookjs/storybook/pull/15871

Telescoping on https://github.com/storybookjs/storybook/pull/15999

## What I did

Changes in the manager to:

 - Fetch `stories.json` and use it to build the sidebar if it is v3 data
 - Deal with the new `STORY_PREPARED` event
 - Add a `useStoryPrepared()` hook for addons to use to show a loading state while they wait for the story to be prepared.

### CODE REVIEW

File: lib/api/src/modules/stories.ts [hunk 1]
```diff
@@ -327,9 +348,53 @@ export const init: ModuleFn = ({
...
       });
     },
+    fetchStoryList: async () => {
+      // This needs some fleshing out as part of the stories list server project
+      const result = await global.fetch('/stories.json');
+      const storyIndex = (await result.json()) as StoryIndex;
+
+      // We can only do this if the stories.json is a proper storyIndex
+      if (storyIndex.v !== 3) {
+        return;
```
Comment (shilman) [inline lib/api/src/modules/stories.ts:358]: Log a warning here at least

Comment (tmeasday) [reply lib/api/src/modules/stories.ts:358]: This happens if you are running a built story book with stories.json on the v6 store. Do we want to log a warning?

Comment (shilman) [reply lib/api/src/modules/stories.ts:358]: I think we should log something, even if it's as simple as `Skipping story index with version v${storyIndex.v}`.

Comment (tmeasday) [reply lib/api/src/modules/stories.ts:358]: OK

**Comment by shilman** [comment]
Merged as part of https://github.com/storybookjs/storybook/pull/15871

**Pull request closed without merging**